### ISSUE TITLE
JsonConvert.SerialiseObject doesn't pass ctor arguments and so validation may throw

### ISSUE THREAD

**Issue by vladsch** [bridgedotnet/Bridge.Newtonsoft.Json#4]
Deserializer should map values from json to single constructor arguments

https://deck.net/6afc6edcae045cf853efbdb762c76099/third
https://dotnetfiddle.net/0LpMqs
```cs
public class Program
{
    public static void Main()
    {
        var x = new MyString("abc");
        var json = JsonConvert.SerializeObject(x);
        var cloneX = JsonConvert.DeserializeObject<MyString>(json);
		
        Console.WriteLine("Original:" + x.Value);
        Console.WriteLine("JSON:" + json);
	Console.WriteLine("Clone:" + cloneX.Value);
		
	var list = new MyList<string>(new[] { "a", "b" });
        json = JsonConvert.SerializeObject(list);
        var cloneList = JsonConvert.DeserializeObject<MyList<string>>(json);
		
	Console.WriteLine();
	Console.WriteLine("Original:" + string.Join(",", list));
        Console.WriteLine("JSON:" + json);
	Console.WriteLine("Clone:" + string.Join(",", cloneList));
    }
}

public class MyString
{
    public MyString(string value)
    {
        if (string.IsNullOrWhiteSpace(value))
        {
            throw new ArgumentException("value");
        }

        Value = value;
    }

    public string Value
    {
        get; private set;
    }
}

public sealed class MyList<T> : IEnumerable<T>
{
    private readonly IEnumerable<T> _values;
    public MyList(IEnumerable<T> values)
    {
        _values = values;
    }

    public IEnumerator<T> GetEnumerator()
    {
        return _values.GetEnumerator();
    }

    IEnumerator IEnumerable.GetEnumerator()
    {
        return GetEnumerator();
    }
}
```

## See Also

* https://forums.bridge.net/forum/bridge-net-pro/bugs/4373-jsonconvert-serialiseobject-doesn-t-pass-ctor-arguments-and-so-validation-may-throw

**Comment by geoffreymcgill** [comment]
A pull request has been submitted with an implementation of this feature. Once the pull request has been reviewed and testing is complete, we will merge into the **master** branch and include with the next release.
